{{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_widget.py

```diff
@@ -1,28 +1,125 @@
 """
-This module is an example of a barebones QWidget plugin for napari
+This module contains four napari widgets declared in
+different ways:
 
-It implements the Widget specification.
-see: https://napari.org/stable/plugins/guides.html?#widgets
+- a pure Python function flagged with `autogenerate: true`
+    in the plugin manifest. Type annotations are used by
+    magicgui to generate widgets for each parameter. Best
+    suited for simple processing tasks - usually taking
+    in and/or returning a layer.
+- a `magic_factory` decorated function. The `magic_factory`
```

Comment:
Should we include a line in the magic_factory bullet point that says "This is recommended for most plugin developers"?